### DISCUSSION TITLE
Updating content via source control or CI/CD

### DISCUSSION THREAD

**Discussion by praveenaj** [outline/outline#3500] [Self Hosting] [answered]
Hi

Can we set up CI/CD and update/push markdown content via source control?

**Comment by tommoor** [comment, marked as answer]
Yes, with one catch – the `documents.update` API endpoint will not update the text content with collaborative editing enabled currently. This will be supported in the near future. https://github.com/outline/outline/issues/3424

**Reply by praveenaj** [reply]
Thanks for your response. But unfortunately I couldn't find any guide which explains updating content from source control (I went through https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/hosting-outline-nipGaCRBDu) . Appreciate if you can point me out to relevant resource.

**Reply by tommoor** [reply]
Ref: [documents.update API documentation](https://www.getoutline.com/developers#tag/Documents/paths/~1documents.update/post)

It would be upto you/your team to figure out how to map this to I'm guessing a git repository? There isn't anything built for this yet

**Reply by praveenaj** [reply]
My bad. I thought it's built-in. Will explore that API. Cheers!
